### PR TITLE
Oauth gherkin specs

### PR DESCRIPTION
**Pull request by davidbishoppp**
This PR includes Gherkin feature files for oauth of Hubspot and Saasquatch. Including high-level and low-level detail of the basic oauth functionality

### CODE REVIEW

**Comment by davidbishoppp** [comment]
[OAuth Gherkin specs](https://trello.com/c/nVJX70Ap/72-oauth-gherkin-specs)

**Comment by MAsimSENG** [comment]
Looks good to me, good work and excellent attention to detail.

**Review by MAsimSENG — APPROVED** [review]

File: features/oauth/high-level/Hubspot-Oauth.feature [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+	@manual
+	Scenario: User completes Hubspot oauth flow with errors
+		Given The user starts the Hubspot oauth flow
+		When The user <Oauth Error>
+		Then The user <Outcome>
+		And The user should not be redirected to the configuration page
+
+	| Oauth Error                     | Outcome                                            |
+	| Closes the pupup window         | Should be shown error text                         |
```
Comment (lucianojd) [inline features/oauth/high-level/Hubspot-Oauth.feature]: Spelling error: pupup looks like it should be popup

**Review by lucianojd — APPROVED** [review]
Left a comment about one spelling mistake. Everything looks good otherwise!

**Pull request merged**